optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/event/SolverEventSupport.java

```diff
@@ -32,19 +32,17 @@
  */
 public class SolverEventSupport<Solution_> extends AbstractEventSupport<SolverEventListener<Solution_>> {
 
-    private Solver<Solution_> solver;
+    private final Solver<Solution_> solver;
 
     public SolverEventSupport(Solver<Solution_> solver) {
         this.solver = solver;
     }
 
     public void fireBestSolutionChanged(DefaultSolverScope<Solution_> solverScope, Solution_ newBestSolution) {
         final Iterator<SolverEventListener<Solution_>> it = eventListenerSet.iterator();
-        long timeMillisSpent = solverScope.calculateTimeMillisSpentUpToNow();
-        Score newBestScore = solverScope.getSolutionDescriptor().getScore(newBestSolution);
         if (it.hasNext()) {
             final BestSolutionChangedEvent<Solution_> event = new BestSolutionChangedEvent<>(solver,
-                    timeMillisSpent, newBestSolution, newBestScore);
+                    solverScope.getBestSolutionTimeMillis(), newBestSolution, solverScope.getBestScore());
```

Comment:
This definitely breaks stuff. spent (since start) != timeMillis (since epoch)... fixing.